src/ShellCheck/Checks/ShellSupport.hs

```diff
@@ -238,15 +242,19 @@ checkBashisms = ForShell [Sh, Dash] $ \t -> do
         warnMsg id "`<file` to read files is"
 
     bashism t@(T_SimpleCommand _ _ (cmd:arg:_))
-        | t `isCommand` "echo" && "-" `isPrefixOf` argString =
+        | t `isCommand` "echo" && hasEchoFlags =
             unless ("--" `isPrefixOf` argString) $ -- echo "-----"
                 if isDash
                 then
                     when (argString /= "-n") $
                         warnMsg (getId arg) "echo flags besides -n"
                 else
                     warnMsg (getId arg) "echo flags are"
-      where argString = concat $ oversimplify arg
+      where
+          argString = concat $ oversimplify arg
+          flagRegex = mkRegex "^-[eEsn]+$"
+          hasEchoFlags = Just [] == matchRegex flagRegex argString
```

Comment:
You can simplify this with the function `matches :: String -> Regex -> Bool`, commonly seen with infix notation:  ``argString `matches` matchRegex``